server/src/main/kotlin/suwayomi/tachidesk/manga/impl/download/DownloadManager.kt

```diff
@@ -80,17 +80,25 @@ object DownloadManager {
     init {
         scope.launch {
             notifyFlow.sample(1.seconds).collect {
-                val status = getStatus()
-                clients.forEach {
-                    it.value.send(status)
-                }
+                sendStatusToAllClients()
             }
         }
     }
 
-    private fun notifyAllClients() {
-        scope.launch {
-            notifyFlow.emit(Unit)
+    private fun sendStatusToAllClients() {
+        val status = getStatus()
+        clients.forEach {
+            it.value.send(status)
+        }
+    }
+
+    private fun notifyAllClients(immediate: Boolean = false) {
+        if (immediate) {
+            sendStatusToAllClients()
+        } else {
+            scope.launch {
+                notifyFlow.emit(Unit)
+            }
         }
```

Comment:
I would suggest doing it this way so that it doesnt block downloads from continueing because of a slow connection to the subscriber.
```suggestion
        if (immediate) {
            scope.launch {
                sendStatusToAllClients()
            }
        } else {
            scope.launch {
                notifyFlow.emit(Unit)
            }
        }
```